src/odemis/driver/pi.py

```diff
@@ -743,6 +745,7 @@ def read(self):
         return self.file.read()
 
     def write(self, arg):
+        arg = arg.decode('latin1') if sys.version_info[0] >= 3 else arg
```

Comment:
If the file is opened in binanry mode, it should accept bytes (and return bytes), which would be more realistic, and would avoid this extra conversion. => `open(..., "bw+")`